webapp/resources/templates/fragments/loginProviders.html

```diff
@@ -22,7 +22,7 @@ <h3 th:utext="#{screen.welcome.label.loginwith}"/>
                     <span th:case="linkedin" class="fa fa-linkedin"></span>
                     <span th:case="facebook" class="fa fa-facebook-square"></span>
                     <span th:case="oidc" class="fa fa-openid"></span>
-                    <span th:case="saml2" class="fa fa-lock"></span>
+                    <span th:case="saml2" th:class="${entry.cssClass}" ></span>
```

Comment:
I think I need the regex.  What I'm trying to do is take an arbitrary string coming from the app configuration and create a valid CSS identifier from it, which I would then tack on to the class using `th:classappend`.  I need to replace all non alphanumeric characters other than - and _ with something (I somewhat arbitrarily chose '-').  Using `String.replaceAll('\\W", '-')`
in Java is a nice simple way to do it, although point taken on the performance issues.

However, for my specific problem, there's no need to compile the regex over and over and over again.  I can easily see how to do this with the in-CAS Java solution (just make the `Pattern` a static part of `DelegatedClientAuthenticationAction.ProviderLoginPageConfiguration` and compile it on init), but I'm not sure how to do it in Thymeleaf.